src/librustc_errors/emitter.rs

```diff
@@ -221,7 +221,9 @@ pub trait Emitter {
                // when this style is set we want the suggestion to be a message, not inline
                sugg.style != SuggestionStyle::HideCodeAlways &&
                // trivial suggestion for tooling's sake, never shown
-               sugg.style != SuggestionStyle::CompletelyHidden
+               sugg.style != SuggestionStyle::CompletelyHidden &&
+               // subtle suggestion, never shown inline
+               sugg.style != SuggestionStyle::ShowAlways
```

Comment:
Now that you have 3 conditions: `![Suggestion::HideCodeAlways, SuggestionStyle::CompletelyHidden, SuggestionStyle::ShowAlways].contains(&sugg.style)`

Reply:
I wanted to keep them separate to spell out the reasoning behind them where it matters, instead of hiding it behind a method call.